tests/distributed/test_topologies/test_topologies.py

```diff
@@ -144,3 +152,56 @@ def test_complex_needs(parallels, mocker):
             on_done=response_mock,
         )
     response_mock.assert_called()
+
+
+@pytest.mark.parametrize('parallel', [1, 2])
+def test_remote_flow_local_executors(mocker, parallel):
+    response_mock = mocker.Mock()
+    client = JinaDClient(host=__default_host__, port=8000)
+    workspace_id = client.workspaces.create(paths=[os.path.join(cur_dir, 'yamls')])
+
+    GATEWAY_LOCAL_GATEWAY = 'flow_glg.yml'
+    GATEWAY_LOCAL_LOCAL_GATEWAY = 'flow_gllg.yml'
+
+    for flow_yaml in [
+        GATEWAY_LOCAL_GATEWAY,
+        GATEWAY_LOCAL_LOCAL_GATEWAY,
+    ]:
+        flow_id = client.flows.create(
+            workspace_id=workspace_id, filename=flow_yaml, envs={'PARALLEL': parallel}
+        )
+        args = client.flows.get(flow_id)['arguments']['object']['arguments']
+        Client(
+            host=__default_host__,
+            port=args['port_expose'],
+            protocol=args['protocol'],
+        ).post(
+            on='/',
+            inputs=(Document(blob=np.random.random([1, 100])) for _ in range(NUM_DOCS)),
+            on_done=response_mock,
+            show_progress=True,
+        )
+        response_mock.assert_called()
```

Comment:
here you should create a new Mock object, right? The first time it can be called but the second time it might not but the assert will still pass, no?